### PR TITLE
Create `Cardano.CLI.Helpers.hs`

### PR DESCRIPTION
**Pull request by Jimbo4350**
Removing some of the clutter from `Cardano.CLI.Run`.

### CODE REVIEW

File: cardano-node/src/Cardano/CLI/Helpers.hs [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+      keyMatchesUTxO :: Crypto.VerificationKey -> CC.UTxO.TxOut -> Maybe CC.UTxO.TxOut
+      keyMatchesUTxO key out =
+        if CC.Common.checkVerKeyAddress key (CC.UTxO.txOutAddress out)
+        then Just out else Nothing
+
+      fromCompactTxInTxOutList :: [(CC.UTxO.CompactTxIn, CC.UTxO.CompactTxOut)]
+                               -> [(CC.UTxO.TxIn, CC.UTxO.TxOut)]
+      fromCompactTxInTxOutList =
+          map (bimap CC.UTxO.fromCompactTxIn CC.UTxO.fromCompactTxOut)
+      cheat     :: CC.UTxO.TxWitness
```
Comment (intricate) [inline cardano-node/src/Cardano/CLI/Helpers.hs]: Unrelated to this PR but, at some point, we should probably come up with a better name for this.

File: cardano-node/src/Cardano/CLI/Helpers.hs [hunk 2]
```diff
@@ -0,0 +1,308 @@
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE RankNTypes                 #-}
+{-# LANGUAGE RecordWildCards            #-}
+
+{-# OPTIONS_GHC -Wno-all-missed-specialisations #-}
```
Comment (intricate) [inline cardano-node/src/Cardano/CLI/Helpers.hs]: Do we need this?

File: cardano-node/src/Cardano/CLI/Run.hs [hunk 3]
```diff
@@ -41,58 +41,35 @@ module Cardano.CLI.Run (
   , NewTxFile(..)
```
Comment (intricate) [inline cardano-node/src/Cardano/CLI/Run.hs]: Are all of the `LANGUAGE` and `OPTIONS_GHC` pragmas still required in this module? (`GeneralisedNewtypeDeriving ` and `-Wno-all-missed-specialisations` in particular)

**Comment by Jimbo4350** [comment]
> Just a few concerns that I wouldn't mind a response to 
> 
> Unrelated to my request for changes:
> 
> I tend to be wary of "util" or "helper" modules since they can end up becoming a dumping ground for things when it becomes a bit more difficult to decide where they belong in the project. However, I do find this to be nicer than it was before so it's fine.
> 
> In the future (assuming we'll be expanding upon the work done in this project) we should give some thought to where these things belong long-term.

Agreed but this is temporary. There was too much clutter in `Cardano.CLI.Run` and as I continue cleaning up `cardano-node` I'll find a better home for these guys.

File: cardano-node/src/Cardano/CLI/Helpers.hs (hunk 2 above)
Comment (Jimbo4350) [reply cardano-node/src/Cardano/CLI/Helpers.hs]: Responded below.

File: cardano-node/src/Cardano/CLI/Run.hs (hunk 3 above)
Comment (Jimbo4350) [reply cardano-node/src/Cardano/CLI/Run.hs]: `-Wno-all-missed-specialisation` is, ghc complains:
```
src/Cardano/CLI/Run.hs: warning:
    Could not specialise imported function ‘cardano-ledger-0.1.0.0:Cardano.Chain.Genesis.Data.$wreadGenesisData’
      when specialising ‘readGenesisData’
    Probable fix: add INLINABLE pragma on ‘cardano-ledger-0.1.0.0:Cardano.Chain.Genesis.Data.$wreadGenesisData
```
And as for `GeneralizedNewtypeDeriving`:
```
src/Cardano/CLI/Run.hs:141:28: error:
    • Can't make a derived instance of ‘IsString NewTxFile’:
        ‘IsString’ is not a stock derivable class (Eq, Show, etc.)
        Try GeneralizedNewtypeDeriving for GHC's newtype-deriving extension
    • In the newtype declaration for ‘NewTxFile’
    |
141 |   deriving (Eq, Ord, Show, IsString)
```

Comment (Jimbo4350) [reply cardano-node/src/Cardano/CLI/Run.hs]: But `{-# OPTIONS_GHC -Wno-partial-fields #-}` needs to go :)

File: cardano-node/src/Cardano/CLI/Helpers.hs (hunk 1 above)
Comment (Jimbo4350) [reply cardano-node/src/Cardano/CLI/Helpers.hs]: Agree with you :100:

File: cardano-node/src/Cardano/CLI/Run.hs (hunk 3 above)
Comment (intricate) [reply cardano-node/src/Cardano/CLI/Run.hs]: Perhaps we can look into whatever fix is required for the `all-missed-specialisations` warning (I haven't looked into it). Probably not worth blocking this PR on.

Comment (intricate) [reply cardano-node/src/Cardano/CLI/Run.hs]: > But `{-# OPTIONS_GHC -Wno-partial-fields #-}` needs to go :)

100%. I see no reason that we should ever suppress this warning.

File: cardano-node/src/Cardano/CLI/Helpers.hs (hunk 1 above)
Comment (deepfire) [reply cardano-node/src/Cardano/CLI/Helpers.hs]: I actually fixed this in https://github.com/input-output-hk/cardano-node/pull/152/files#diff-75f92b9083d544707ede537871a2a640L438

Are you sure you rebased properly, @Jimbo4350 ?

Comment (deepfire) [reply cardano-node/src/Cardano/CLI/Helpers.hs]: I actually removed that name in https://github.com/input-output-hk/cardano-node/pull/152/files#diff-75f92b9083d544707ede537871a2a640L438

Are we sure we had a clean rebase, @Jimbo4350 @intricate?

File: cardano-node/src/Cardano/CLI/Run.hs (hunk 3 above)
Comment (deepfire) [reply cardano-node/src/Cardano/CLI/Run.hs]: Again, @intricate, @Jimbo4350, this was fixed in https://github.com/input-output-hk/cardano-node/pull/152/files#diff-75f92b9083d544707ede537871a2a640L21

Did we have a clean rebase?

**Comment by deepfire** [comment]
@Jimbo4350, @intricate, https://github.com/input-output-hk/cardano-node/pull/162#discussion_r320041073 makes me wonder -- what other pieces of https://github.com/input-output-hk/cardano-node/pull/152 were undone.

**Comment by deepfire** [comment]
I bet Tx submission no longer works.

**Comment by deepfire** [comment]
> In the future (assuming we'll be expanding upon the work done in this project) we should give some thought to where these things belong long-term.

@intricate, I also had slightly different plans about restructuring this code.

I also find it somewhat sad that Jordan never talked to me about this module reorganisation.

Never even mentioned it once.

File: cardano-node/src/Cardano/CLI/Helpers.hs (hunk 1 above)
Comment (intricate) [reply cardano-node/src/Cardano/CLI/Helpers.hs]: I went over the content of this PR but I'm unsure about changes in relation to #152.

Comment (intricate) [reply cardano-node/src/Cardano/CLI/Helpers.hs]: In fact, I didn't look very closely at #152. Only a couple parts where I was mentioned (and maybe some other minor things I noticed like the formatting thing).

**Comment by Jimbo4350** [comment]
> @intricate, I also had slightly different plans about restructuring this code.
> 
> I also find it somewhat sad that Jordan never talked to me about this module reorganisation.
> 
> Never even mentioned it once.

@deepfire I requested your review on this PR 4 days ago.

**Comment by deepfire** [comment]
> I requested your review on this PR 4 days ago. Fixes pushed to jordan/rebase-fix.

@Jimbo4350, are you serious?

The four days include today (which I don't even know how you would theorize it qualifies), sunday and saturday. and the PR was open on Friday's evening.

Given how much change it introduces, how about at least making sure I'm aware of it, like you did with many other PRs?

**Pull request merged**